src/hooks/use-ordered-rows.ts

```diff
@@ -30,6 +32,17 @@ export function useOrderedRows<Row>(
         return 0;
       }
 
+      // We want null/undefined elements to be initially placed at the bottom of
+      // the list, but still move to the top if order direction changes.
+      // We check a/b instead of x/y because nulls should not be affected by the
+      // regular order direction.
+      if (a === null || a === undefined) {
+        return initialOrder.current?.direction === order.direction ? 1 : -1;
+      }
+      if (b === null || b === undefined) {
+        return initialOrder.current?.direction === order.direction ? -1 : 1;
+      }
```

Comment:
I think these two conditions should also evaluate that `initialOrder.current?.field === order.field`. I'm trying to verify it.

```suggestion
      if ((a === null || a === undefined) && initialOrder.current?.field === order.field) {
        return initialOrder.current?.direction === order.direction ? 1 : -1;
      }
      if ((b === null || b === undefined) && initialOrder.current?.field === order.field) {
        return initialOrder.current?.direction === order.direction ? -1 : 1;
      }
```

Reply:
Actually, I think the right thing to do is not to track `initialOrder` as a ref to `order`, but instead, keep a ref to a map of the initial order per field. That way each "column" will always start with its null values at the bottom.

Reply:
Yep, this was the right approach ☝🏼